torch/_inductor/fx_passes/overlap_preserving_bucketer.py

```diff
@@ -356,8 +367,14 @@ def _find_buckets(
         buckets = []
         processed: OrderedSet[fx.Node] = OrderedSet()
 
-        # Sort collectives by node index for efficient distance checking
-        sorted_collectives = sorted(collective_group, key=lambda n: self.node_idx[n])
+        # Sort by overlap ratio (ascending) to bucket least hidden collectives first.
+        # Exposed collectives benefit most from bucketing since their latency is on the
+        # critical path. Prioritizing them also preserves hiding relationships for
+        # already-hidden collectives, which have less to gain from bucketing.
+        sorted_collectives = sorted(
```

Comment:
Hmm, could we do something where we sort first by exposure, but then try to bucket locally ? it feels a bit unintuitive to try to bucket from the start of the graph till the end.

See also:

```
            # Greedy optimization: stop after consecutive failures
            consecutive_failures = 0
            max_consecutive_failures = 20

            # Check candidates in sorted order, break when beyond max distance
            for candidate in sorted_collectives[i + 1 : i + 1 + self.max_coll_distance]:
```